### PR TITLE
Add flake8 checks and fix errors

### PR DESCRIPTION
**Pull request by thomas-riccardi**
Add `flake8`:
- add flake8 settings
- add flake8 call in travis-ci
- more verbose tests in travis-ci
- fix flake8 errors
- improve .gitignore

Also:
- add CONTRIBUTING.md
- store dev requirements in requirements.dev.txt
- test before package
- add test coverage report in travis-ci logs

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -10,11 +10,16 @@ branches:
 
 # command to install dependencies
 install:
-  - pip install -r requirements.txt
-  - python setup.py bdist_wheel
-  - pip install dist/deepomatic_cli-*.whl
-  - pip install pytest==4.3.0
+  - pip install -r requirements.dev.txt wheel
```
Comment (maingoh) [inline .travis.yml]: maybe put wheel in `requirements.dev.txt` ?

File: deepomatic/cli/cli_parser.py [hunk 2]
```diff
@@ -71,29 +76,62 @@ def argparser_init():
     subparsers_dict['add_images'] = add_images_parser
 
     # Define argument groups for easier reading
-    input_groups = {cmd: subparsers_dict[cmd].add_argument_group('Input parameters') for cmd in ['infer', 'draw', 'blur', 'noop', 'add_images']}
-    output_groups = {cmd: subparsers_dict[cmd].add_argument_group('Output parameters') for cmd in ['infer', 'draw', 'blur', 'noop']}
-    model_groups = {cmd: subparsers_dict[cmd].add_argument_group('Model parameters') for cmd in ['infer', 'draw', 'blur']}
-    onprem_groups = {cmd: subparsers_dict[cmd].add_argument_group('On-premises parameters') for cmd in ['infer', 'draw', 'blur']}
-    option_groups = {cmd: subparsers_dict[cmd].add_argument_group('Option parameters') for cmd in ['infer', 'draw', 'blur', 'noop', 'add_images']}
+    input_groups = {
```
Comment (maingoh) [inline deepomatic/cli/cli_parser.py]: not super fan of those big comprehension, but they were already there ..

File: deepomatic/cli/output_data.py [hunk 3]
```diff
@@ -1,3 +1,5 @@
+# -*- coding: utf-8 -*-
```
Comment (maingoh) [inline deepomatic/cli/output_data.py]: is it necessary ?

File: requirements.txt [hunk 4]
```diff
@@ -4,4 +4,4 @@ requests>=2.19.0,<3  # will not work below in python3
 text-unidecode==1.2
 gevent==1.4.0
 jsonschema==3.0.1
-deepomatic-api==0.9.1
\ No newline at end of file
+deepomatic-api==0.9.1
```
Comment (maingoh) [inline requirements.txt]: what changed ?

Comment (thomas-riccardi) [reply requirements.txt]: maybe end of file `\n` ?

File: deepomatic/cli/cli_parser.py (hunk 2 above)
Comment (thomas-riccardi) [reply deepomatic/cli/cli_parser.py]: yep.. I searched for pep8 formatting recommendations, and this seemed to be the least worse.

File: deepomatic/cli/output_data.py (hunk 3 above)
Comment (thomas-riccardi) [reply deepomatic/cli/output_data.py]: There are non ascii characters in a comment (file tree view) and it somehow failed a python 2.7 travis ci build with precisely that issue. I have no idea why it didn't fail before though.

Comment (thomas-riccardi) [reply deepomatic/cli/output_data.py]: https://travis-ci.com/Deepomatic/deepocli/jobs/262624370

**Comment by thomas-riccardi** [comment]
Code coverage report in travis ci logs (in the middle):
```
---------- coverage: platform linux2, python 2.7.15-final-0 ----------
Name                                                  Stmts   Miss  Cover
-------------------------------------------------------------------------
deepomatic/__init__.py                                    6      6     0%
deepomatic/cli/__init__.py                                4      0   100%
deepomatic/cli/__main__.py                                8      8     0%
deepomatic/cli/cli_parser.py                            114      9    92%
deepomatic/cli/cmds/__init__.py                           0      0   100%
deepomatic/cli/cmds/feedback.py                          63     17    73%
deepomatic/cli/cmds/infer.py                            149     77    48%
deepomatic/cli/cmds/studio_helpers/__init__.py            0      0   100%
deepomatic/cli/cmds/studio_helpers/file.py              118     32    73%
deepomatic/cli/cmds/studio_helpers/task.py               19      4    79%
deepomatic/cli/cmds/studio_helpers/vulcan2studio.py      42     22    48%
deepomatic/cli/common.py                                 40     13    68%
deepomatic/cli/exceptions.py                             41      7    83%
deepomatic/cli/frame.py                                  23      5    78%
deepomatic/cli/input_data.py                            337    109    68%
deepomatic/cli/json_schema.py                            14      0   100%
deepomatic/cli/output_data.py                           265    133    50%
deepomatic/cli/thread_base.py                           264     63    76%
deepomatic/cli/version.py                                 7      0   100%
deepomatic/cli/workflow/__init__.py                      22      2    91%
deepomatic/cli/workflow/cloud_workflow.py                49     11    78%
deepomatic/cli/workflow/json_workflow.py                 38      8    79%
deepomatic/cli/workflow/rpc_workflow.py                  86     54    37%
deepomatic/cli/workflow/workflow_abstraction.py          21      7    67%
-------------------------------------------------------------------------
TOTAL                                                  1730    587    66%
```

File: deepomatic/cli/cli_parser.py (hunk 2 above)
Comment (maingoh) [reply deepomatic/cli/cli_parser.py]: You would actually need to use subvariables to do something clearer and/or do a normal loop. It's okay like that.

File: deepomatic/cli/output_data.py (hunk 3 above)
Comment (maingoh) [reply deepomatic/cli/output_data.py]: It probably comes from flake (which parses the comments). Python2 just ignore them.

File: requirements.txt (hunk 4 above)
Comment (maingoh) [reply requirements.txt]: It is usually displayed in the diff, weird !

File: deepomatic/cli/output_data.py (hunk 3 above)
Comment (thomas-riccardi) [reply deepomatic/cli/output_data.py]: it fails in pytest, after flake8 successfully finishing... it was also before I add pytest-cov to the requirements.dev.txt... the mystery deepens.

**Pull request merged**